### ISSUE TITLE
New problem after #63

### ISSUE THREAD

**Issue by michael3lyb** [FormidableLabs/react-native-app-auth#67]
The fix in #63 (v2.3.0) works well on fitbit. But after I switched back to [Jawbone API](https://jawbone.com/up/developer/account), it failed again with "failed exchange token".  I tested Jawbone API on (v2.2.1), it worked well. I think the fix in #63 generates some new issues.

**Comment by jevakallio** [comment]
@michael3lyb can you share example configuration for the Jawbone API?

**Comment by michael3lyb** [comment]
Sure. Sorry forgot that. 

```
const jawboneConfig = {
  clientId: '*******',
  clientSecret: '****************************************',
  redirectUrl: 'com.example://redirect', 
  scopes: ['basic_read', 'location_read'], 
  serviceConfiguration: {
    authorizationEndpoint: 'https://jawbone.com/auth/oauth2/auth',
    tokenEndpoint: 'https://jawbone.com/auth/oauth2/token',
  }
};
```

**Comment by cwcooper** [comment]
I'm having the same issue with the Meetup.com API.  This is my first attempt to hook this up in React Native, but everything is working fine when I try at https://developers.google.com/oauthplayground

`
  const config = {
      clientId: '*************',
      clientSecret: '******************',
      redirectUrl: '***********://redirect',
      scopes: ['basic'],
      serviceConfiguration: {
        authorizationEndpoint: 'https://secure.meetup.com/oauth2/authorize',
        tokenEndpoint: 'https://secure.meetup.com/oauth2/access'
      }
    };
`

`
Error: Failed exchange token
    at createErrorFromErrorData (NativeModules.js:123)
    at NativeModules.js:80
    at MessageQueue.__invokeCallback (MessageQueue.js:347)
    at MessageQueue.js:137
    at MessageQueue.__guard (MessageQueue.js:265)
    at MessageQueue.invokeCallbackAndReturnFlushedQueue (MessageQueue.js:136)
    at debuggerWorker.js:72
`

**Comment by cheft** [comment]
me too

**Comment by SteffiPeTaffy** [comment]
I am having the same issue as described in #63 trying to authenticate with strava. RNAppAuthModule rejects the promise because of "org.json.JSONException: field "token_type" not found in json object".

**Comment by kadikraman** [comment]
Closing this due to age (the issue references v2.3.0 and we're now on v5).

Please open a new issue if this problem persist on the latest version.
